torchaudio/datasets/vctk.py

```diff
@@ -136,3 +139,93 @@ def __getitem__(self, n: int) -> Tuple[Tensor, int, str, str, str]:
+    Each item is a tuple of the form: (waveform, sample_rate, utterance, speaker_id, utterance_id)
+    Folder `p315` will be ignored due to the non-existent corresponding text files.
+    For more information about the dataset visit: https://datashare.is.ed.ac.uk/handle/10283/3443
+    """
+
+    def __init__(self,
+                 root: str,
+                 url: str = URL,
+                 download: bool = False,
+                 mic_id: str = 'mic2') -> None:
+
+        archive = os.path.join(root, os.path.basename('VCTK-Corpus-0.92.zip'))
+
+        self._path = os.path.join(root, 'VCTK-Corpus-0.92')
+        self._txt_dir = os.path.join(self._path, 'txt')
+        self._audio_dir = os.path.join(self._path, 'wav48_silence_trimmed')
+
+        if download:
+            if not os.path.isdir(self._path):
+                if not os.path.isfile(archive):
+                    checksum = _CHECKSUMS.get(url, None)
+                    download_url(url, root, hash_value=checksum, hash_type="md5")
+                extract_archive(archive, self._path)
+
+        if not os.path.isdir(self._path):
+            raise RuntimeError(
+                "Dataset not found. Please use `download=True` to download it."
+            )
+
+        # Extracting speaker IDs from the folder structure
+        self._speaker_ids = sorted(os.listdir(self._txt_dir))
+        self._sample_ids = []
+
+        """
+        Due to some insufficient data complexity in the 0.92 version of this dataset,
+        we start traversing the audio folder structure in accordance with the text folder.
+        As some of the audio files are missing of either ``mic_1`` or ``mic_2`` but the
+        text is present for the same, we first check for the existence of the audio file
+        before adding it to the ``sample_ids`` list.
+
+        Once the ``audio_ids`` are loaded into memory we can quickly access the list for
+        different parameters required by the user.
+        """
+        for speaker_id in self._speaker_ids:
+            utterance_dir = os.path.join(self._txt_dir, speaker_id)
+            for utterance_file in sorted(f for f in os.listdir(utterance_dir) if f.endswith('.txt')):
+                utterance_id = os.path.splitext(utterance_file)[0]
+                audio_path_mic = os.path.join(self._audio_dir, speaker_id,
+                                              f'{utterance_id}_{mic_id}.flac')
+                if speaker_id == 'p280' and mic_id == 'mic2':
+                    break
+                elif speaker_id == 'p362':
+                    if os.path.isfile(audio_path_mic):
+                        self._sample_ids.append(f'{utterance_id}_{mic_id}')
+                else:
+                    self._sample_ids.append(f'{utterance_id}_{mic_id}')
+
+    def load_text(self, file_path) -> str:
+        with open(file_path) as file_path:
+            return file_path.readlines()[0]
+
+    def load_audio(self, file_path) -> Tuple[Tensor, int]:
```

Comment:
Could you append this method name with underscore `_` so as to tell this is not expected to be used publicly??

Reply:
Done